crates/pagecache/src/config.rs

```diff
@@ -515,43 +516,46 @@ impl Config {
 
     // panics if config options are outside of advised range
     fn validate(&self) -> Result<(), ()> {
-        supported!(
-            self.inner.io_bufs <= 32,
+        check_supported!(
+            self.inner.io_bufs > 32,
             "too many configured io_bufs. please make <= 32"
         );
-        supported!(
-            self.inner.io_buf_size >= 100,
-            "io_buf_size should be hundreds of kb at minimum, and we won't start if below 100"
+        check_supported!(
+            self.inner.io_buf_size < 100,
+            "io_buf_size should be hundreds of kb at minimum, and we won't \
+            start if below 100"
         );
-        supported!(
-            self.inner.io_buf_size <= 1 << 24,
+        check_supported!(
+            self.inner.io_buf_size > 1 << 24,
             "io_buf_size should be <= 16mb"
         );
-        supported!(self.inner.page_consolidation_threshold >= 1, "must consolidate pages after a non-zero number of updates");
-        supported!(self.inner.page_consolidation_threshold < 1 << 20, "must consolidate pages after fewer than 1 million updates");
-        supported!(
-            self.inner.cache_bits <= 20,
+        check_supported!(self.inner.page_consolidation_threshold < 1,
+            "must consolidate pages after a non-zero number of updates");
+        check_supported!(self.inner.page_consolidation_threshold >= 1 << 20,
+            "must consolidate pages after fewer than 1 million updates");
+        check_supported!(
+            self.inner.cache_bits > 20,
             "# LRU shards = 2^cache_bits. set cache_bits to 20 or less."
         );
-        supported!(
-            self.inner.segment_cleanup_threshold >= 0.01,
+        check_supported!(
+            self.inner.segment_cleanup_threshold < 0.01,
             "segment_cleanup_threshold must be >= 1%"
         );
-        supported!(
-            self.inner.segment_cleanup_skew < 99,
+        check_supported!(
+            self.inner.segment_cleanup_skew >= 99,
             "segment_cleanup_skew cannot be greater than 99%"
         );
-        supported!(
-            self.inner.compression_factor >= 1,
-            "compression_factor must be >= 0"
+        check_supported!(
+            self.inner.compression_factor < 1,
+            "compression_factor must be >= 1"
         );
-        supported!(
-            self.inner.compression_factor <= 22,
+        check_supported!(
+            self.inner.compression_factor > 22,
             "compression_factor must be <= 22"
         );
-        supported!(
-            self.inner.idgen_persist_interval > 0,
-            "idgen_persist_interval must be above 0"
+        check_supported!(
+            self.inner.idgen_persist_interval < 1,
```

Comment:
<=

Reply:
whoops, nvm

Reply:
is "supported" a clippy warning? I think I would prefer to keep the conditionals the way they were

Reply:
Not supported specifically, I can change the word back. Here's the clippy error: https://rust-lang.github.io/rust-clippy/master/index.html#neg_cmp_op_on_partial_ord. So to solve it I inverted the condition. This way it also follows flow similar to an if statement, which is why I added the word "check" to reinforce that it is a condition. I can change the name back if you'd like. Any thoughts on avoiding inverting the condition, if that's an issue?

Reply:
I'd prefer to keep it more like an assert, where the condition must be true, and supported to stay the same. For that specific error, feel free to throw an outer `!` around the PartialOrd positive comparison